test/unit/math/prim/meta/is_eigen_test.cpp

```diff
@@ -1,57 +0,0 @@
-#include <stan/math/prim.hpp>
-#include <Eigen/Sparse>
-#include <gtest/gtest.h>
-#include <vector>
-
-TEST(MathMetaPrim, primitive) {
-  using stan::is_eigen;
-  EXPECT_FALSE((is_eigen<bool>::value));
-  EXPECT_FALSE((is_eigen<double>::value));
-  EXPECT_FALSE((is_eigen<int>::value));
-
-  EXPECT_FALSE((is_eigen<std::vector<double>>::value));
-
-  EXPECT_TRUE((is_eigen<Eigen::EigenBase<Eigen::MatrixXd>>::value));
-  EXPECT_TRUE((is_eigen<Eigen::Matrix<double, -1, -1>>::value));
-  EXPECT_TRUE((is_eigen<Eigen::SparseMatrix<double>>::value));
-  EXPECT_TRUE((is_eigen<Eigen::MatrixBase<Eigen::MatrixXd>>::value));
-
-  EXPECT_TRUE((is_eigen<const Eigen::Matrix<double, -1, -1>>::value));
-  EXPECT_TRUE((is_eigen<Eigen::SparseMatrix<double>&>::value));
-  EXPECT_TRUE(
-      (is_eigen<Eigen::MatrixBase<Eigen::Matrix<double, -1, -1>>&&>::value));
-
-  Eigen::Matrix<double, -1, -1> a;
-  Eigen::Matrix<double, -1, -1> b;
-  Eigen::VectorXd c;
-
-  EXPECT_TRUE((is_eigen<decltype(a * b)>::value));
-  EXPECT_TRUE((is_eigen<decltype(a * b + a.transpose())>::value));
-  EXPECT_TRUE((is_eigen<decltype(c.segment(0, 1))>::value));
```

Comment:
This check was added after a bug was found. It must be included in the new testing framework.

Reply:
Yes good catch. I'll include tests for this and block()